src/server/services/Actions.ts

```diff
@@ -37,35 +45,51 @@ export default class Actions {
       waitReady().then(() => {
         this.account = keyring.addFromMnemonic(mnemonic);
 
-        setInterval(() => {
-          // We do want the following to just start and run
-          // eslint-disable-next-line @typescript-eslint/no-floating-promises
-          // TODO: Adding a subscription would be better but the server supports on http for now
-          this.updateFaucetBalance().catch(console.error);
-        }, balancePollIntervalMs);
+        this.updateFaucetBalance().then(() =>
+          logger.info('Fetched faucet balance 💰')
+        );
+        // We do want the following to just start and run
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        // TODO: Adding a subscription would be better but the server supports on http for now
+        setInterval(() => this.updateFaucetBalance(), balancePollIntervalMs);
```

Comment:
`setInterval` isn't a good idea because it can end up spamming the server with "stacked" requests if they end up overlapping (e.g. due to delays). It's better to reschedule the call only when the previous `updateFaucetBalance` finishes, i.e.

```js
const updateFaucetBalance = () => {
  this.updateFaucetBalance().then(() =>
    setTimeout(updateFaucetBalance, balancePollIntervalMs),
  )
}
updateFaucetBalance()
```

Reply:
Good point, thank you!